app/lib/frontend/templates.dart

```diff
@@ -402,8 +400,32 @@ class TemplateService {
             documentationUrl.startsWith('http://pub.dartlang.org/'))) {
       documentationUrl = null;
     }
-    final isGitHubHomepage = selectedVersion.homepage != null &&
-        selectedVersion.homepage.startsWith('https://github.com/');
+    final isGitHubHomepage =
+        homepageUrl != null && homepageUrl.startsWith('https://github.com/');
+    final dartdocsUrl = urls.pkgDocUrl(
+      package.name,
+      version: selectedVersion.version,
+      isLatest: selectedVersion.version == package.latestVersion,
+    );
+    final repositoryUrl = urls.inferRepositoryUrl(homepageUrl);
+    final issueTrackerUrl = urls.inferIssueTrackerUrl(homepageUrl);
+    final repositoryIsHomepage = repositoryUrl == homepageUrl;
+
+    final links = <Map<String, dynamic>>[];
+    void addLink(String href, String label) {
+      if (href == null || href.isEmpty) return;
+      links.add(<String, dynamic>{'href': href, 'label': label});
+    }
+
+    if (repositoryIsHomepage) {
+      addLink(repositoryUrl, 'Repository');
+    } else {
+      addLink(homepageUrl, isGitHubHomepage ? 'Homepage (GitHub)' : 'Homepage');
+      addLink(repositoryUrl, 'Repository');
+    }
+    addLink(issueTrackerUrl, 'Issue Tracker');
```

Comment:
'Issue tracker' is maybe not as clear as 'Bug reports' or 'Report a bug in Foo' or something.

For github, we could preformat a message like we do with our own 'report a bug' link that shows the pub URl the person came from.